src/appservice/AccessControl.ts

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { Bridge } from "matrix-appservice-bridge";
+import AccessControlUnit, { EntityAccess } from "../models/AccessControlUnit";
+import PolicyList from "../models/PolicyList";
+import { Permalinks } from "matrix-bot-sdk";
+
+// We need to refactor AccessControlUnit so you can have
+// previousAccess and currentAccess listener for changes.
+// wait that only works for literals not globs...
+// i guess when the rule change is a glob we have to scan everything.
+export class AccessControl {
+
+    private constructor(
+        private readonly accessControlList: PolicyList,
+        private readonly accessControlUnit: AccessControlUnit
+        ) {
+    }
+
+    public static async setupAccessControl(
```

Comment:
It's a static method because we don't want someone to contruct the access control without initializing it properly and we can't use async in the constructor. So instead the contructor is private. 

>Why do we have both a PolicyList as a member instance and one returned by a static method?

I can't see that here?

Reply:
Well, we have one [here](https://github.com/matrix-org/mjolnir/pull/364/files#diff-2a0a9fe6c6d3c7b5b6e9a419d03251ec1e237a5d0371158c6002d1cb07d9bf42R29) and a different one [here](https://github.com/matrix-org/mjolnir/pull/364/files#diff-2a0a9fe6c6d3c7b5b6e9a419d03251ec1e237a5d0371158c6002d1cb07d9bf42R42-R46). I don't understand what each is for.

Reply:
Have documented